### PR TITLE
Fixed the wallet loading

### LINKED ISSUE #778: Tx list is garbled when wallet loaded
## Actual behaviour
If my wallet has Sigma spends and I open my wallet when Sigma is disabled, I can see that the tx list is not what it was before. Tx times have changed, tx order has changed

## Desired behaviour
Tx list is not changed.

## Steps to reproduce
- regtest, mine 400, mint, spend and mint again
- mine up to 511, close wallet, open - the tx list will be garbled.

### PR DESCRIPTION
**Pull request by a-bezrukov**
Fixes #778

### CODE REVIEW

File: src/sigma.cpp [hunk 1]
```diff
@@ -185,13 +185,21 @@ bool CheckSigmaSpendTransaction(
...
         bool isCheckWallet,
         bool fStatefulSigmaCheck,
         CSigmaTxInfo *sigmaTxInfo) {
     bool hasSigmaSpendInputs = false, hasNonSigmaInputs = false;
     int vinIndex = -1;
     std::unordered_set<Scalar, sigma::CScalarHash> txSerials;
 
+    Consensus::Params const & params = ::Params().GetConsensus();
+
+    if(!isVerifyDB && !isCheckWallet) {
```
Comment (riordant) [inline src/sigma.cpp]: Just noticed that `isVerifyDB` is always passed as false to this function. Is it necessary to keep it?

File: src/sigma.cpp [hunk 2]
```diff
@@ -421,15 +428,14 @@ bool CheckSigmaTransaction(
         realHeight = chainActive.Height();
     }
 
-    if(tx.IsSigmaSpend() && realHeight >= consensus.nDisableUnpaddedSigmaBlock && realHeight < consensus.nSigmaPaddingBlock)
-        return state.DoS(100, error("Sigma is bisabled at this period."));
+    bool const allowSigma = (realHeight >= consensus.nSigmaStartBlock);
```
Comment (riordant) [inline src/sigma.cpp]: Shouldn't we just return immediately if Sigma is not allowed? (I understand that this code already existed)

File: src/sigma.cpp (hunk 1 above)
Comment (levonpetrosyan93) [reply src/sigma.cpp]: it is code cleanup, better do it with another PR I think.

Comment (a-bezrukov) [reply src/sigma.cpp]: It was my initial implementation but I was asked to make the change smaller.

File: src/sigma.cpp (hunk 2 above)
Comment (a-bezrukov) [reply src/sigma.cpp]: We may be checking the wallet transactions or the block DB transactions. These checks should be done on Sigma disabled block heights.

**Pull request merged**